### PR TITLE
C-compile compatibility for udMath.h, udPlatform.h and udResult.h

### PR DESCRIPTION
**Pull request by dpev**
Paves the way for removal of udGeometryMath etc

### CODE REVIEW

File: Source/udJSON.cpp [hunk 1]
```diff
@@ -186,94 +186,101 @@ udResult udJSON::SetObject()
...
-      break;
-  }
-epilogue:
-  return result;
+  return udJSON_SetArrayHelper(this, pArray, length, shrinkMathTypes);
 }
 
+//// ****************************************************************************
+//// Author: Dave Pevreal, May 2017
+//udResult udJSON::Set(const udDouble4x4 &v, bool shrink)
```
Comment (samsinsane) [inline Source/udJSON.cpp:243]: Delete perhaps?

File: Source/udPlatform.cpp
Comment (samsinsane) [inline Source/udPlatform.cpp:1]: Also doesn't need to change.

File: Source/udResult.cpp
Comment (samsinsane) [inline Source/udResult.cpp:1]: Also doesn't need to change.

File: Source/udSocket.cpp [hunk 2]
```diff
@@ -1,3 +1,4 @@
+#define NOMINMAX // Required on Windows in order to use std::min/std::max from <algorithm>
```
Comment (samsinsane) [inline Source/udSocket.cpp:1]: Not required if put into `udPlatform.h`

File: Source/udStringUtil.cpp [hunk 3]
```diff
@@ -1,11 +1,13 @@
+#define NOMINMAX // Required on Windows in order to use std::min/std::max from <algorithm>
```
Comment (samsinsane) [inline Source/udStringUtil.cpp:1]: Same as above.

File: Source/udUUID.cpp [hunk 4]
```diff
@@ -1,9 +1,10 @@
+#define NOMINMAX
```
Comment (samsinsane) [inline Source/udUUID.cpp:1]: Same as above.

File: Source/udWorkerPool.cpp [hunk 5]
```diff
@@ -1,13 +1,14 @@
+#define NOMINMAX
```
Comment (samsinsane) [inline Source/udWorkerPool.cpp:1]: Same as above.

File: udTest/src/udCompressionTests.cpp [hunk 6]
```diff
@@ -21,12 +20,12 @@ TEST(udCompressionTests, Basic)
     EXPECT_EQ(udR_Success, udCompression_Deflate(&pDeflated, &deflatedSize, input, UDARRAYSIZE(input), compressionType));
     EXPECT_NE(input, pDeflated);
     EXPECT_NE(UDARRAYSIZE(input), deflatedSize);
-    EXPECT_NE(0, memcmp(input, pDeflated, udMin(UDARRAYSIZE(input), deflatedSize)));
+    EXPECT_NE(0, memcmp(input, pDeflated, min(UDARRAYSIZE(input), deflatedSize)));
```
Comment (samsinsane) [inline udTest/src/udCompressionTests.cpp:23]: If `NOMINMAX` is put into `udPlatform.h` the usages in this file will need to be updated to `std::`

**Review by samsinsane — CHANGES_REQUESTED** [review]

**Comment by dpev** [comment]
Closing in favour of staging commits differently

**Pull request closed without merging**